src/dfm/setup.rs

```diff
@@ -0,0 +1,25 @@
+use super::model::Config;
+use anyhow::{Result, anyhow};
+use std::fs;
+use std::path::PathBuf;
+
+pub fn setup(path: PathBuf) -> Result<()> {
+    let config = generate_config(path)
+
+    println!("{:?}", config?);
+
+    Ok(())
+}
+
+fn generate_config(path: PathBuf) -> Result<Config> {
+    let config_content: String = fs::read_to_string(&path).expect("");
+
+    let extension = path.extension().expect("").to_str().unwrap();
```

Comment:
```suggestion
    let extension = path.extension().extension().and_then(|e| e.to_str()).context("Failed to convert extension to str")?;
```
So -- both `Option::expect` and `Option::unwrap` will panic. This is kinda a weird operation because both `extension` and `to_str` return `Option` wrapped types. You could handle both options independently:

```rust
let extension = extension().context("Failed to get extension")?.to_str().context("Failed to convert extension to str")?
```

But that felt a little verbose for a single operation. Instead using the `Option::and_then` method you can chain Option-returning functions which will unwrap when `Option::Some(x)` into `x` and end the chain when `None`. This means you can process both `Options` at the end and add `context` to convert them into a `Result` type which allows you to use the `?` operator and return early on error.